snapcraft/tests/__init__.py

```diff
@@ -139,6 +139,13 @@ def setUp(self):
         self.parts_dir = os.path.join(os.getcwd(), 'parts')
         self.local_plugins_dir = os.path.join(self.snap_dir, 'plugins')
 
+        machine = os.environ.get('MOCK_MACHINE', None)
```

Comment:
Would it make sense to try to namespace the environment variables? Something like SNAPCRAFT_TEST_MOCK_MACHINE.

Here are some env vars that only apply to travis, like $DEPENDENCIES and $TEST_SUITE, I'm not sure if those should also be namespaced. But $TEST_SNAP_WITH_TRACKS I think should be something like $SNAPCRAFT_TEST_SNAP_WITH_TRACKS. If we agree on a style, I can apply it in a later branch.